### PR TITLE
feat: add to calendar button on meetup page

### PR DESCRIPTION
**Pull request by damevin**
# What does this MR do?

This merge request introduces the 'Add to Calendar' feature, allowing users to easily add upcoming meetups to their personal calendars.

Screenshot-ydOpKONe095822

#92

### CODE REVIEW

**Comment by MrSunshyne** [comment]
Thanks for the PR @damevin ! 

I deployed the preview, you can see it by clicking the "View deployment" button in the environemnt section on this page.
There's an error where it's trying to import from node_modules, which isn't available on prod. 

It appears that you need to add the `hoist` attribute on the script tag, it will bundle that script and ship it with the build (Today I Learned 😅)

```diff
- <script type="module">
+ <script type="module" hoist>
```

Could you also move all the code for add to calendar into it's own component ? We're revamping this page very soon and having the logic for this feature in a a dedicated file will make it much nicer to relocate it [in the upcoming dock ](https://frontendmu-astro-7kp1hlwiy-frontendmu.vercel.app/meetup/50)

**Comment by damevin** [comment]
Oh yes @MrSunshyne you're right! I'm sorry this is the first time with Astro for me and add dependencies by using intersection observer sounds weird for me 😅. 

I'll do a fix tomorrow night if it's ok for you

**Comment by damevin** [comment]
@MrSunshyne please take a look when you have time 😇

File: src/components/MeetupSingle.astro [hunk 1]
```diff
@@ -7,13 +7,14 @@ import MeetupGallery from "./MeetupGallery.astro";
 import SessionList from "./SessionList.astro";
 import { photoAlbumSource } from "src/config";
 import photosResponse from "@data/photos-raw.json";
-import { vTransitionName } from "@utils/helpers";
+import { vTransitionName, convertTo24HourFormat } from "@utils/helpers";
```
Comment (MrSunshyne) [inline src/components/MeetupSingle.astro:10]: This import is no longer required in the current file

**Review by MrSunshyne — APPROVED** [review]

**Comment by MrSunshyne** [comment]
> Oh yes @MrSunshyne you're right! I'm sorry this is the first time with Astro for me and add dependencies by using intersection observer sounds weird for me 😅.
> 

Absolutely fine ! I also learned a new pattern. It's pretty cool actually, the dependency is not loaded unless that component is visible. 

The changes look good to me ! merging now.

I noticed that the library does some "advertising" in the console. I'm not a big fan of that, but we can keep it for now

image

**Pull request merged**